statsmodels/tsa/arima_model.py

```diff
@@ -968,9 +968,8 @@ def fit(self, start_params=None, trend='c', method="css-mle",
         self.transparams = False  # so methods don't expect transf.
 
         normalized_cov_params = None  # TODO: fix this
-        armafit = ARMAResults(self, params, normalized_cov_params)
-        armafit.mle_retvals = mlefit.mle_retvals
-        armafit.mle_settings = mlefit.mle_settings
+        armafit = ARMAResults(self, params, normalized_cov_params,
+                              method=method, mlefit=mlefit)
```

Comment:
AFAICS, `mlefit=mlefit` keeps the super model alive her, but it is not attached in `__init__` and so will still go out of scope.
i.e. LGTM